apps/rush/src/actions/GenerateAction.ts

```diff
@@ -59,13 +58,13 @@ export default class GenerateAction extends CommandLineAction {
     const stopwatch: Stopwatch = Stopwatch.start();
     const isLazy: boolean = this._lazyParameter.value;
 
-    console.log('Starting "rush generate"' + os.EOL);
-
-    if (this._rushConfiguration.packageReviewFile) {
-      this._packageReviewChecker = new PackageReviewChecker(this._rushConfiguration);
-      this._packageReviewChecker.saveCurrentDependencies();
+    if (this._rushConfiguration.approvedPackagesPolicyEnabled) {
+      const packageReviewChecker: PackageReviewChecker = new PackageReviewChecker(this._rushConfiguration);
+      packageReviewChecker.saveCurrentDependencies();
     }
```

Comment:
@nickpape-msft That seems like overkill.  Plus, you want to carefully control when it happens in the sequence.

What I could do is roll it up into a concise static function call like `PackageReviewChecker.saveDependencies(this._rushConfiguration);`.  Eventually I want to improve this design, e.g. to validate-without-rewriting.  But that's out of scope for this PR, which is mainly about finalizing file formats.

---
In reply to: [116121866](https://github.com/Microsoft/web-build-tools/pull/181#discussion_r116121866) [](ancestors = 116121866)